utils/src/cas.rs

```diff
@@ -0,0 +1,450 @@
+// Copyright (C) 2022 Alibaba Cloud. All rights reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+use std::fmt::{self, Display, Formatter};
+use std::io;
+use std::path::{Path, PathBuf};
+
+use rusqlite::{Connection, OptionalExtension};
+
+use crate::digest::RafsDigest;
+
+/// Error codes related to local cas.
+#[derive(Debug)]
+pub enum CasError {
+    Io(std::io::Error),
+    Db(rusqlite::Error),
+}
+
+impl Display for CasError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
+impl std::error::Error for CasError {}
+
+impl From<rusqlite::Error> for CasError {
+    fn from(e: rusqlite::Error) -> Self {
+        CasError::Db(e)
+    }
+}
+
+impl From<io::Error> for CasError {
+    fn from(e: io::Error) -> Self {
+        CasError::Io(e)
+    }
+}
+
+/// Specialized `Result` for local cas.
+type Result<T> = std::result::Result<T, CasError>;
+
+pub struct CasMgr {
+    conn: Connection,
+}
+
+impl CasMgr {
+    pub fn new(path: impl AsRef<Path>) -> Result<CasMgr> {
+        let mut db_path = path.as_ref().to_owned();
+        db_path.push("cas.db");
+        let conn = Connection::open(db_path)?;
+
+        // create blob and chunk table for nydus v5
+        conn.execute(
+            "CREATE TABLE IF NOT EXISTS BlobInfos_V5 (
+            BlobId     TEXT NOT NULL PRIMARY KEY,
+            BlobInfo   TEXT NOT NULL,
+            Backend    TEXT NOT NULL
+        )",
+            (),
+        )?;
+        conn.execute(
+            "CREATE INDEX IF NOT EXISTS BlobIndex_V5 ON BlobInfos_V5(BlobId)",
+            (),
+        )?;
+
+        conn.execute(
+            "CREATE TABLE IF NOT EXISTS ChunkInfos_V5 (
+            ChunkId           INTEGER PRIMARY KEY,
+            ChunkDigestValue  TEXT NOT NULL,
+            ChunkInfo         TEXT NOT NULL,
+            BlobId            TEXT NOT NULL,
```

Comment:
Should we use an auto-incr integer as primary key for BlobInfos_V5, so we can store an integer instead of a text here.